### DISCUSSION TITLE
Why isn't ThemeProvider in React Material UI (5.0.4) not working with Storybook?

### DISCUSSION THREAD

**Discussion by roysatadru** [storybookjs/storybook#16451] [Help] [answered]
Am I missing something here guys? Please guide me - "How can I fix this issue"?

My code in preview.js:
```
import { ThemeProvider, createTheme, CssBaseline } from '@mui/material';

export const parameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
};

const theme = createTheme({
  palette: {
    primary: {
      main: '#373fc5',
    },
    secondary: {
      main: '#fba428',
    },
    background: {
      default: '#161616',
      paper: '#232323',
    },
    divider: '#353535',
  },
  shape: {
    borderRadius: '0.1875rem',
  },
  typography: {
    fontFamily: ['"Lato"', 'sans-serif'].join(','),
    h1: {
      fontFamily: '"Arimo", sans-serif',
      fontSize: '1.60rem',
      textTransform: 'uppercase',
      fontWeight: 700,
      color: '#fba428',

      '@media only screen and (min-width: 430px)': {
        fontSize: '1.85rem',
        fontWeight: 400,
      },
    },
    h2: {
      fontFamily: '"Arimo", sans-serif',
      fontSize: '0.7rem',
      textTransform: 'uppercase',
      fontWeight: 400,
    },
    body1: {
      color: '#e0e0e0',
      fontSize: '0.8rem',
      letterSpacing: '0.02rem',
    },
    body2: {
      color: '#828282',
      fontSize: '0.6rem',
      letterSpacing: '0.05rem',
      margin: '0.9rem 0',
      lineHeight: 1.7,
    },
    button: {
      fontFamily: '"Fira Sans", sans-serif',
      fontWeight: 800,
      letterSpacing: '0.05rem',
      fontSize: '0.8rem',
    },
  },
  components: {
    MuiButton: {
      defaultProps: {
        disableRipple: true,
      },
      styleOverrides: {
        contained: {
          height: '2.5rem',
          padding: '0 2rem',
          boxShadow: '0 0 0.625rem 0 rgba(0, 0, 0, 0.05)',
          transition: 'all 300ms ease-in',
          lineHeight: '2.5rem',
          verticalAlign: 'middle',
        },
        containedPrimary: {
          '&:hover': {
            color: '#000',
            backgroundColor: '#fba428',
            boxShadow: '0 0 1.25rem 0 rgba(0, 0, 0, 0.2)',
          },
        },
      },
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          boxShadow:
            '0 0 0.625rem 0 rgb(76 87 125 / 16%), 0 0.09375rem 0.125rem 0 rgba(255, 255, 255, 0.15) inset',
        },
      },
    },
  },
});

export const decorators = [
  Story => (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Story />
    </ThemeProvider>
  ),
];
```

My code in Button.tsx:
```
import { forwardRef } from 'react';
import { ButtonProps, Button as MuiButton } from '@mui/material';

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => <MuiButton ref={ref} {...props} />,
);
```

My code in Button.stories.tsx:
```
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Button } from './Button';

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    onClick: { control: 'onClick' },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = args => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: 'Click Me!',
  variant: 'contained',
  color: 'primary',
};
```

I have also replaced the Global Decorator used in this code with both Component Decorator & Story Decorator. But it still doesn't work and throws a weird error. The error might be stemming from the Theme not getting applied properly. Following image is the error that shows itself whenever I switch to the "Docs" tab. Please guide me how I can fix the issue.

![image](https://user-images.githubusercontent.com/64960866/138483958-625b9b63-96e2-412f-9b83-d72e4a42619b.png)

**Comment by shilman** [comment, marked as answer]
Storybook docs page renders stories inline by default and Storybook's emotion version is conflicting with MUI5's version. We have plans to fix this but unfortunately it's not straightforward.

In the meantime, you should be able to work around this by disabling  inline rendering in docs mode:

https://storybook.js.org/docs/react/writing-docs/docs-page#inline-stories-vs-iframe-stories

Or you can disable the docs addon entirely (assuming you're using addon essentials):

https://storybook.js.org/docs/react/essentials/introduction#disabling-addons

**Reply by roysatadru** [reply]
Thank you @shilman for the answer. Really appreciate it.

But that's not the only problem, right? I mean, the theming of MUI5 also doesn't work. So if I want to test out the components with the themes applied, I can't do that.

Yeah this answer definitely can help me not show the error on the docs page. But we will be basically hiding the problem that we are facing, right?

But yeah, you have plans to fix it. Good luck for that! So, for now, we should be going with MUI4 to access all of the cool features, it provides with storybook, right?

And one more thing, should I raise this discussion as an issue as well?

**Reply by shilman** [reply]
Please give this a try & let me know if it works (and if we can improve the documentation):

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility

**Comment by jacksonblankenship** [comment]
I'm also having trouble with this. I can't get the theme to apply at all. It seems like all of the examples online use `@material-ui/core` but now we're working with `@mui/material`. Is there an issue/PR I can track to keep an eye on this?

**Reply by shilman** [reply]
Please give this a try & let me know if it works (and if we can improve the documentation):

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility

**Reply by roysatadru** [reply]
@jacksonblankenship tell me if the suggested solution works. If it doesn't, I will raise this as an issue.

**Comment by bastiengrignon** [comment]
Hello, I've got the same issue but I'm not using `emotion` but `styled-components` instead ! I checked the workaround and it seems to be only for `emotion` and disabling **docs addons** or **essentials addons** doesn't work as proposed by @shilman. 

I'd like to know if there is a solution for the moment ?
